src/controller/deviceindex/ShipDeviceIndexMappingManager.cpp

```diff
@@ -409,6 +409,8 @@ void ShipDeviceIndexMappingManager::HandlePhysicalDeviceDisconnectSinglePlayer(i
 }
 
 void ShipDeviceIndexMappingManager::HandlePhysicalDeviceDisconnectMultiplayer(int32_t sdlJoystickInstanceId) {
+    auto portIndexOfPhysicalDeviceThatHasBeenDisconnected =
+        GetPortIndexOfDisconnectedPhysicalDevice(sdlJoystickInstanceId);
```

Comment:
`GetShip` closes the controller

https://github.com/Kenix3/libultraship/blob/80c2c5e9c6a5681428f759100c2428c5e41f333c/src/controller/deviceindex/ShipDeviceIndexMappingManager.cpp#L546

so by the time this was being called 

https://github.com/Kenix3/libultraship/blob/80c2c5e9c6a5681428f759100c2428c5e41f333c/src/controller/deviceindex/ShipDeviceIndexMappingManager.cpp#L481

`mController` would already be null

https://github.com/Kenix3/libultraship/blob/80c2c5e9c6a5681428f759100c2428c5e41f333c/src/controller/controldevice/controller/mapping/sdl/SDLMapping.cpp#L145-L146

meaning it always didn't know what device was disconnected

moving this up here makes it not a problem